lib/boom_notifier/error_storage.ex

```diff
@@ -4,7 +4,7 @@ defmodule BoomNotifier.ErrorStorage do
   # Keeps track of the errors grouped by type and a counter so the notifier
   # knows the next time it should be executed
 
-  use Agent
+  use Agent, start: {__MODULE__, :start_link, []}, type: :supervisor
```

Comment:
Keeping it as a `type: :supervisor` for the scope of this PR because that was the existing behaviour with `Supervisor.Spec.supervisor/2` call, but does this agent really need to be a supervisor or it's fine as a worker?

Reply:
I don't think we need the `type: :supervisor` at all so I'm fine with removing it (which will default to `type: :worker`)

Reply:
Cool, I can do that as a quick separate follow-up PR so we keep this PR as a mere refactor without behavior changes if that works.

Reply:
...motivation for keeping in separate commits is so that we keep the ability later of reverting the behavior change and keeping the refactor in case some unexpected problem with the change.

Reply:
Yeah, I'm ok with a follow up PR for changing this

Reply:
https://github.com/wyeworks/boom/pull/49